pkgs/development/libraries/gmock/default.nix

```diff
@@ -1,15 +1,13 @@
-{ stdenv, fetchurl, unzip, cmake}:
-
+{ stdenv, cmake, callPackage }:
+let
+  source = callPackage ./source.nix { };
```

Comment:
That was a partial misunderstanding between us. It's enough to just write
```nix
let
  source = fetchzip { /* ... */ };
```
I don't think there's any benefit in making a `mkDerivation` wrapper around that to only copy the same thing into another path. If you want to override `name`, that can also be passed to `fetchzip` directly.